dashboard/src/components/LoginForm/LoginForm.tsx

```diff
@@ -19,7 +22,15 @@ interface ILoginFormState {
 
 class LoginForm extends React.Component<ILoginFormProps, ILoginFormState> {
   public state: ILoginFormState = { token: "" };
+
+  public componentDidMount() {
+    this.props.tryToAutoAuthenticate();
+  }
+
   public render() {
+    if (this.props.autoAuthenticating) {
```

Comment:
Sorry, I wasn't very clear, I was actually suggesting that we reuse `this.props.authenticating` and show a loading screen instead of the form. This does mean that when logging in with an invalid token, it'll flash to a loading screen and then back to the form, but I think that's not too bad and most users entering tokens correctly won't see any issues.

Reply:
ah, okay, that makes more sense